Amplify/Core/Support/AsyncChannel.swift

```diff
@@ -0,0 +1,88 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+actor AsyncChannel<Element: Sendable>: AsyncSequence {
+    struct Iterator: AsyncIteratorProtocol, Sendable {
+        private let channel: AsyncChannel<Element>
+
+        init(_ channel: AsyncChannel<Element>) {
+            self.channel = channel
+        }
+
+        mutating func next() async -> Element? {
+            await channel.next()
```

Comment:
Check for cancellation? https://developer.apple.com/documentation/swift/asynciteratorprotocol

Reply:
What can be canceled here?